cl/search/management/commands/generate_opinion_embeddings.py

```diff
@@ -0,0 +1,182 @@
+            create_opinion_text_embeddings.si(batch, database).set(
+                queue=embedding_queue
+            ),
+            save_embeddings.s().set(queue=upload_queue),
+        ).apply_async()
+
+    def handle(self, *args, **options):
+        embedding_queue = options["embedding_queue"]
+        upload_queue = options["upload_queue"]
+        database = options["database"]
+        batch_size = options["batch_size"]
+        count = options.get("count", None)
+        auto_resume = options["auto_resume"]
+        min_opinion_size = settings.MIN_OPINION_SIZE
+        start_id = options["start_id"]
+        throttle_min_items = options["throttle_min_items"]
+        self.throttle = CeleryThrottle(
+            queue_name=embedding_queue, min_items=throttle_min_items
+        )
+        if auto_resume:
+            start_id = get_last_parent_document_id_processed(
+                compose_redis_key()
+            )
+            self.stdout.write(
+                f"Auto-resume enabled starting embedding from ID: {start_id}."
+            )
+
+        opinions = Opinion.objects.filter(id__gte=start_id)
+        # Limit opinions to retrieve if count was provided.
+        opinions_to_process = (
+            opinions[:count] if count is not None else opinions
+        )
+        opinions_with_best_text = opinions.order_by("pk").with_best_text()
+        opinions_with_best_text = (
+            opinions_with_best_text[:count] if count is not None else opinions
+        )
+
+        count = opinions_to_process.count()
+        current_batch: list[int] = []
+        current_batch_size = 0
+        processed_count = 0
+        for opinion in opinions_with_best_text.iterator():
+            opinion_id = opinion.pk
+            processed_count += 1
+            token_count = opinion.token_count
+            if token_count < min_opinion_size:
+                continue
+            if token_count > batch_size:
+                # Log documents that individually exceed the batch size.
+                logger.error(
+                    "The opinion ID:%s exceeds the batch size limit.",
+                    opinion_id,
+                )
+                continue
+            # Check if adding this opinion would exceed the batch size.
+            if current_batch_size + token_count > batch_size:
+
+                # Send the current batch since adding this opinion would break the limit.
+                self.send_batch(
+                    current_batch, embedding_queue, upload_queue, database
+                )
+                current_batch = []
+                current_batch_size = 0
+
+            current_batch.append(opinion_id)
+            current_batch_size += token_count
+            logger.info(
+                "\rProcessed %s/%s, (%s), last ID requested for embedding: %s",
+                processed_count,
+                count,
+                f"{processed_count * 1.0 / count:.0%}",
+                opinion_id,
+            )
```

Comment:
This is going to make 10M log lines, isn't it?

Reply:
Fixed it. Now this is logged every 1000 items